src/controllers/restful/PaymentController.ts

```diff
@@ -0,0 +1,204 @@
+/**
+ * PaymentControllers
+ * RESTful Controllers for Payments Endpoints
+ */
+
+import * as express from "express";
+import { Either, left, right } from "fp-ts/lib/Either";
+import { PPTPortTypes } from "italia-pagopa-api/dist/wsdl-lib/PagamentiTelematiciPspNodoservice/PPTPort";
+import * as uuid from "uuid";
+import { PagoPaConfig } from "../../Configuration";
+import { ControllerError } from "../../enums/ControllerError";
+import { HttpErrorStatusCode } from "../../enums/HttpErrorStatusCode";
+import * as PaymentsService from "../../services/PaymentsService";
+import { PaymentsActivationRequest } from "../../types/controllers/PaymentsActivationRequest";
+import { PaymentsActivationResponse } from "../../types/controllers/PaymentsActivationResponse";
+import { PaymentsCheckRequest } from "../../types/controllers/PaymentsCheckRequest";
+import { PaymentsCheckResponse } from "../../types/controllers/PaymentsCheckResponse";
+import { CodiceContestoPagamento } from "../../types/PagoPaTypes";
+import * as PaymentsConverter from "../../utils/PaymentsConverter";
+import * as RestfulUtils from "../../utils/RestfulUtils";
+// Forward a payment check request from BackendApp to PagoPa
+export async function checkPaymentToPagoPa(
+  req: express.Request,
+  res: express.Response,
+  pagoPaConfig: PagoPaConfig
+): Promise<Either<ControllerError, PaymentsCheckResponse>> {
+  // Validate input
+  const errorOrPaymentsCheckRequest = PaymentsCheckRequest.decode(req.params);
+  if (errorOrPaymentsCheckRequest.isLeft()) {
+    return left(
+      RestfulUtils.sendErrorResponse(
+        res,
+        ControllerError.ERROR_INVALID_INPUT,
+        HttpErrorStatusCode.BAD_REQUEST
+      )
+    );
+  }
+
+  // Generate a session token (this is the first message of a payment flow)
+  const errorOrCodiceContestoPagamento = generateCodiceContestoPagamento();
+  if (errorOrCodiceContestoPagamento.isLeft()) {
+    return left(
+      RestfulUtils.sendErrorResponse(
+        res,
+        ControllerError.ERROR_INTERNAL,
+        HttpErrorStatusCode.INTERNAL_ERROR
+      )
+    );
+  }
+
+  // Convert controller request to PagoPa request
```

Comment:
I can see that we're converting the objects, what I'd like to know, reading the code, is: what types we're converting to what other types and how/why and what part of the pagoPa flow is this?